File: order.service/src/main/java/br/com/edu/order/domain/repository/OrderRepository.java
/* (C)2024 */
package br.com.edu.order.domain.repository;

import br.com.edu.order.domain.Order;
import java.util.UUID;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface OrderRepository extends JpaRepository<Order, UUID> {

    Page<Order> findAll(Pageable pageable);

}

File: order.service/order-service-usecase/src/main/kotlin/br/com/edu/order/repository/OrderRepository.kt
package br.com.edu.order.repository

import br.com.edu.order.domain.Order
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import java.util.*

interface OrderRepository {

    fun findAll(pageable: Pageable): Page<Order>

    fun findById(id: UUID): Optional<Order>

    fun findByIdForUpdate(id: UUID): Optional<Order>

    fun save(order: Order): Order

    fun deleteAll()

    fun count(): Long
}
